Toolkit/ArcGISToolkit/AR/ArcGISARView.swift

```diff
@@ -0,0 +1,531 @@
+            ])
+    }
+    
+    /// Start the locationManager with undetermined access.
+    fileprivate func startWithAccessNotDetermined() {
+        if (Bundle.main.object(forInfoDictionaryKey: "NSLocationWhenInUseUsageDescription") != nil) {
+            locationManager.requestWhenInUseAuthorization()
+        }
+        if (Bundle.main.object(forInfoDictionaryKey: "NSLocationAlwaysUsageDescription") != nil) {
+            locationManager.requestAlwaysAuthorization()
+        }
+        else{
+            didStartOrFailWithError(ArcGISARView.missingPListKeyError())
+        }
+    }
+    
+    /// Start updating the location and heading via the locationManager.
+    fileprivate func startUpdatingLocationAndHeading() {
+        locationManager.startUpdatingLocation()
+        if CLLocationManager.headingAvailable() {
+            locationManager.startUpdatingHeading()
+        }
+    }
+    
+    /// Stop updating location and heading.
+    fileprivate func stopUpdatingLocationAndHeading() {
+        locationManager.stopUpdatingLocation()
+        if CLLocationManager.headingAvailable() {
+            locationManager.stopUpdatingHeading()
+        }
+    }
+
+    /// Start the locationManager with denied access.
+    fileprivate func startWithAccessDenied() {
+        didStartOrFailWithError(ArcGISARView.accessDeniedError())
+    }
+    
+    /// Start the locationManager with authorized access.
+    fileprivate func startWithAccessAuthorized() {
+        startUpdatingLocationAndHeading()
+    }
+    
+    /// Potential notification to the user of an error starting device tracking.
+    ///
+    /// - Parameter error: The error that occurred when starting tracking.
+    fileprivate func didStartOrFailWithError(_ error: Error?) {
+        if !notifiedStartOrFailure, let error = error {
+            // TODO: present error to user...
+            print("didStartOrFailWithError: \(String(reflecting:error))")
+        }
+        
+        notifiedStartOrFailure = true
+    }
+    
+    /// Handle a change in authorization status to "denied".
+    fileprivate func handleAuthStatusChangedAccessDenied() {
+        // auth status changed to denied
+        stopUpdatingLocationAndHeading()
+        
+        // We were waiting for user prompt to come back, so notify.
+        didStartOrFailWithError(ArcGISARView.accessDeniedError())
+    }
+    
+    /// Handle a change in authorization status to "authorized".
+    fileprivate func handleAuthStatusChangedAccessAuthorized() {
+        // Auth status changed to authorized; now that we have authorization - start updating location and heading.
+        didStartOrFailWithError(nil)
+        startUpdatingLocationAndHeading()
+    }
+    
+    /// Called when device orientation changes.
+    ///
+    /// - Parameter notification: The notification.
+    @objc func orientationChanged(notification: Notification?) {
+        // Handle rotation here.
+        switch UIApplication.shared.statusBarOrientation {
```

Comment:
A couple things I just noticed:
- You're subscribing to `UIDevice.orientationDidChangeNotification`, but you're checking `UIApplication.statusBarOrientation`. Perhaps you mean to check `UIDevice.orientation` or instead subscribe to `UIApplication. didChangeStatusBarOrientationNotification`?
- `UIApplication.statusBarOrientation` is deprecated.

Reply:
Fixed; using UIDevice.current.orientation, but I will look into the other things I mentioned.